README.md

```diff
@@ -58,32 +124,312 @@ client = Redis.new(semian: {
-   connect method with one that wraps the connect call with
-   `Semian::Resource#acquire`. You should do this at the lowest possible level.
-4. `query`. Same as `connect` but for queries on the resource.
-5. Define exceptions `ResourceBusyError` and `CircuitOpenError`. These are
-   raised when the request was rejected early because the resource is out of
-   tickets or because the circuit breaker is open (see [Understanding
-   Semian](#understanding-semian). They should inherit from the base exception
-   class from the raw driver. For example `Mysql2::Error` or
-   `Redis::BaseConnectionError` for the MySQL and Redis drivers. This makes it
-   easy to `rescue` and handle them gracefully in application code, by
-   `rescue`ing the base class.
+## Do I need Semian?
 
-The best resource is looking at the [already implemented adapters](#adapters).
+Semian is not trivial library to understand, introduces complexity and thus
+should be introduced with care. Remember, all Semian does is raise exceptions
+based on heuristics. It is paramount that you understand Semian before
+including it in production as you may otherwise be surprised by its behaviour.
+
+Applications that benefit from Semian are those working on eliminating SPOFs,
+and specifically are running into a wall regarding slow resources. But it is by
+no means a magic wand that solves all your latency problems by being added to
+your `Gemfile`. This section describes the types of problems Semian solves.
+
+If your server isn't single threaded (e.g. Resque and Unicorn) or evented these
+problems are less immediate. However, still recommended at scale.
+
+### Real World Example
+
+This is better illustrated with a real world example from Shopify. If you are
+browsing a store and you are signed in, Shopify stores your session in a session
+store. If the session store becomes unavailable, the data driver will start
+throwing driver errors. We rescue these exceptions and simply disable all
+customer sign in functionality on the store until the session store is back
+online.
+
+This is great if querying the resource fails instantly, because it means we fail
+in just a single inter-dc roundtrip of ~1ms. But if the resource is
+unresponsive, or slow, this can take as long as our timeout which is easily
+200ms. This means every request, even if it does rescue the exception, now
+takes an extra 200ms. Because every resource takes that long, our capacity is
+also significantly degraded. These problems are explained in depth in the next
+two sections.
+
+With Semian, the slow resource would fail instantly (after a small amount of
+convergence time) preventing your response time from spiking and not decreasing
+capacity of the cluster.
+
+If this sounds familiar to you, Semian is what you need to be resilient to
+latency. You may not need the graceful fallback depending on your application,
+in which case it will just result in an error (e.g. a `HTTP 500`) quicker which
```

Comment:
I'd replace "quicker" with "faster" and get rid of the "usually good thing" bit.

> in which case it will just result in an error (e.g. a `HTTP 500`) faster.